src/c++/perf_analyzer/profile_data_exporter.cc

```diff
@@ -245,6 +249,39 @@ ProfileDataExporter::AddVersion(std::string& raw_version)
   document_.AddMember("version", version, document_.GetAllocator());
 }
 
+void
+ProfileDataExporter::AddServiceKind(cb::BackendKind& kind)
+{
+  std::string raw_service_kind{""};
+  if (kind == cb::BackendKind::TRITON) {
+    raw_service_kind = "triton";
+  } else if (kind == cb::BackendKind::TENSORFLOW_SERVING) {
+    raw_service_kind = "tfserving";
+  } else if (kind == cb::BackendKind::TORCHSERVE) {
+    raw_service_kind = "torchserve";
+  } else if (kind == cb::BackendKind::TRITON_C_API) {
+    raw_service_kind = "triton_c_api";
+  } else if (kind == cb::BackendKind::OPENAI) {
+    raw_service_kind = "openai";
+  } else {
+    std::cerr << "Unknown service kind detected. The 'service_kind' will not "
+                 "be specified."
+              << std::endl;
+  }
+
+  rapidjson::Value service_kind;
+  service_kind.SetString(raw_service_kind.c_str(), document_.GetAllocator());
+  document_.AddMember("service_kind", service_kind, document_.GetAllocator());
+}
+
+void
+ProfileDataExporter::AddEndpoint(std::string& raw_endpoint)
+{
+  rapidjson::Value endpoint;
+  endpoint = rapidjson::StringRef(raw_endpoint.c_str());
```

Comment:
Why use StringRef here but SetString for service kind?

Reply:
StringRef was not working for the service kind case. Not sure why tho.

Reply:
I think it's because the lifetime of the variable holding the service kind expired after the AddServiceKind function finishes, so when the JSON is actually serialized later, it's using a bad memory address.

It'd be nice to have consistency though, but we can just skip for now.